src/firmware/boards/powerboard_esp32/PowerLoop.cpp

```diff
@@ -0,0 +1,275 @@
+
+                Serial.println(hv_data);
+
+                delay(500); //wait 0.5s
+             
+            }
+
+            EEPROM.write(HV_MEASUREMENT_ADDRESS, hv_data);
+
+        }
+
+        if(MESSAGE == GENEVA_ENCODER){
+
+            //Action
+            g_encoder_angle = analogRead(G_ENCODER_PIN);
+
+            Serial.println(g_encoder_angle);
+
+            EEPROM.write(G_ENCODER_ADDRESS, g_encoder_angle);
+        }
+
+        if(MESSAGE == GENEVA_MOTOR_ANGLE){
+
+            //Action
+            byte checker = SUART.available(); //check if a character has arrived via SUART Port
+            if (checker != 0) //a charctaer has arrived; it has been auto saved in FIFO; say 1 as 0x31
+            {
+                g_motor_angle = SUART.read(); //read arrived character from FIFO (say 1) and put into x as 0x31 
+
+                analogWrite(G_MOTOR_ANGLE_PIN, g_motor_angle); //Set motor angle
+            }
+            else{
+                Serial.println("Failed Retrieve Geneva Motor Angle");
+            }
+
+
+            EEPROM.write(G_MOTOR_ANGLE_ADDRESS, g_motor_angle);
+        }
+
+        if(MESSAGE == GENEVA_MOTOR_SPEED){
+
+            //Action
+            checker = SUART.available(); //check if a character has arrived via SUART Port
+            if (checker != 0) //a charctaer has arrived; it has been auto saved in FIFO; say 1 as 0x31
+            {
+                g_motor_speed = SUART.read(); //read arrived character from FIFO (say 1) and put into x as 0x31 
+
+                analogWrite(G_MOTOR_SPEED_PIN, g_motor_speed); //Set motor angle
+            }
+            else{
+                Serial.println("Failed Retrieve Geneva Motor Speed");
+            }
+
+            EEPROM.write(G_MOTOR_SPEED_ADDRESS ,g_motor_speed);
+        }
+
+        if(MESSAGE == BREAKBEAM){
+
+            //Action
+            breakbeam_inter = analogRead(BREAKBEAM_DIFF_PIN);
+            Serial.println(breakbeam_inter);
+
+            breakbeam_diff = analogRead(BREAKBEAM_INTER_PIN);
+            Serial.println(breakbeam_diff);
+
+            EEPROM.write(BREAKBEAM_INTER_ADDRESS, breakbeam_inter);
+            EEPROM.write(BREAKBEAM_DIFF_ADDRESS, breakbeam_diff);
+        }
+
+        if(MESSAGE == CHICKER_AUTO_KICK){
+
+            //Action
+            checker = SUART.available(); //check if a character has arrived via SUART Port
+            if (checker != 0) //a charctaer has arrived; it has been auto saved in FIFO; say 1 as 0x31
+            {
+                autokick = SUART.read(); //read arrived character from FIFO (say 1) and put into x as 0x31 
+
+                analogWrite(CHICKER_AUTOKICK_PIN, autokick); //Set motor angle
+            }
+            else{
+                Serial.println("Failed Retrieve Chicker Auto Kick Data");
+            }
+
+            EEPROM.write(CHICKER_AUTOKICK_ADDRESS, autokick);
+        }
+
+        if(MESSAGE == CHICKER_AUTO_CHIP){

```

Comment:
see comments above on line 86

but generally it seems like these it statements are stubbed out with a readpin/log

id recommend writing a function for each readable sensor and each "writable" actuator

then in the loop you can go 

~~~
   // Request Handling
   request = getIncomingJetsonNanoRequestMessageHereCOBS(); (Cody is working on this)
   
   if (request.kick_angle != current_geneva_angle)
   {
       moveGeneva(request.kick_angle);
   }
   
   if (request.kick_speed != 0 && request.autokick == false)
   {
         kick(request.kick_speed); // internally this function would trigger the appropriate pulse width, handle cooldowns etc
   }
  ...

   // Status Response
   struct PowerLoopStatus status;
   
   status.measurement_volts = getHVMeasurement(); // we right these functions and stub them out
   status.breakbeam_tripped = getBreakBeam();
   status.geneva_angle = getGenevaAngle();
   status.battery_voltage = getBatteryVoltage();
   status.current_draw = getCurrentDraw();
   ...
   
    sendPowerLoopStatusToJetsonNanoOverCOBS(status); // Cody is working on this
~~~